cachi2/core/models/output.py

```diff
@@ -19,6 +19,7 @@ class Component(pydantic.BaseModel):
 
     name: str
     version: Optional[str]
+    properties: list[dict[str, str]] = [{"name": "cachi2:foundBy", "value": "cachi2"}]
```

Comment:
1. can we have a Property model?
2. do we want our properties to be camelCase?
3. when we want to add a non-static property from the package manager code, we will override this default and lose the foundBy

Reply:
Maybe the name should also be capitalized consistently, it's `cachi2` here but `Cachi2` in tools

TBH I prefer all lowercase, but it's already Cachi2 everywhere

Reply:
Although maybe we can use `cachi2` in code and Cachi2 in documentation

Reply:
I think we can keep as lowercase "cachi2" at least within the SBOM.

Reply:
2. I copied that from Syft without a second thought. I think the spec suggests it to use hyphens: https://github.com/CycloneDX/cyclonedx-property-taxonomy

Reply:
3. Not sure how to improve this. Maybe we can add this "found-by" to all components as part of the RequestOutput creation, wdyt?

Reply:
I don't think the spec suggests any naming style, even the `cdx` namespace itself is not consistent (mostly [camelCase](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/device.md#cdxdevice-namespace-taxonomy), some kebab-case ([1](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/device.md#cdxdevicegs1-namespace-taxonomy), [2](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/npm.md#cdxnpmpackageconstraint-namespace-taxonomy)), sometimes literal [env variable names](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/gomod.md#cdxgomodbuildenv-namespace-taxonomy))

I would choose something and stick to it, personally prefer snake_case

Reply:
> we can add this "found-by" to all components as part of the RequestOutput creation, wdyt?

depends on what you mean by RequestOutput creation :smile: 

One way could be a properties `@validator` that appends the found_by

Reply:
> I would choose something and stick to it, personally prefer snake_case
I don't mind snake_case. Unless someone else has a different opinion, let's stick to that.